# **Install the following Libraries with pip. Not required for Google Colab.**

*   numpy
*   pandas
*   sklearn
*   xgboost


# **Importing necessary Libraries**

In [27]:
import numpy as np 
import pandas as pd
import os, sys
from sklearn.preprocessing import MinMaxScaler
from xgboost import XGBClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

# **Read the data by using Pandas Library exported in the previous step and display first 5 records. Data can be downloaded from this [link](https://archive.ics.uci.edu/ml/machine-learning-databases/parkinsons/). File name is parkinson.data**

In [28]:
df=pd.read_csv('parkinsons.data') #Read the file
df.head() # Display first five records

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
0  phon_R01_S01_1      119.992       157.302  ...  0.266482  2.301442  0.284654
1  phon_R01_S01_2      122.400       148.650  ...  0.335590  2.486855  0.368674
2  phon_R01_S01_3      116.682       131.111  ...  0.311173  2.342259  0.332634
3  phon_R01_S01_4      116.676       137.871  ...  0.334147  2.405554  0.368975
4  phon_R01_S01_5      116.014       141.781  ...  0.234513  2.332180  0.410335

[5 rows x 24 columns]

# **Descriptive statistics of the dataset by using describe method.**
Since most of the fields are numeric excpet 'name', by default it will skip the name column and show only numeric fields**

In [38]:
df.describe()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...          D2         PPE
count   195.000000    195.000000  ...  195.000000  195.000000
mean    154.228641    197.104918  ...    2.381826    0.206552
std      41.390065     91.491548  ...    0.382799    0.090119
min      88.333000    102.145000  ...    1.423287    0.044539
25%     117.572000    134.862500  ...    2.099125    0.137451
50%     148.790000    175.829000  ...    2.361532    0.194052
75%     182.769000    224.205500  ...    2.636456    0.252980
max     260.105000    592.030000  ...    3.671155    0.527367

[8 rows x 23 columns]

# **Featurs (all columns except 'status') and Labels ('status' Column)**

In [29]:
features=df.loc[:,df.columns!='status'].values[:,1:]
labels=df.loc[:,'status'].values

# **'status' columns consists of 0 and 1. Next step is to display the count of both 0 and 1**

In [30]:
print(labels[labels==1].shape[0], labels[labels==0].shape[0])

147 48


# **147 values are 1 and 48 values are zero in the 'status' column**


---



# **Time to scale the features and normalise them i.e. their values will be in between -1 and 1. It will be done by using MinMaxScaler.**
# **fit_transform method is used to fit and transform the data**

In [31]:
scaler=MinMaxScaler((-1,1))
x=scaler.fit_transform(features)
y=labels

# **Next Step is to split the dataset by using train_test_split**.
# **Test size is kept at 20% of the dataset**

In [32]:
x_train, x_test, y_train, y_test=train_test_split(x, y, test_size=0.2, random_state=7)

# **Next Step is to Initialize the XGB Classifier and train the model.**

In [33]:
model=XGBClassifier()
model.fit(x_train, y_train)

XGBClassifier(base_score=0.5, booster='gbtree', colsample_bylevel=1,
              colsample_bynode=1, colsample_bytree=1, gamma=0,
              learning_rate=0.1, max_delta_step=0, max_depth=3,
              min_child_weight=1, missing=None, n_estimators=100, n_jobs=1,
              nthread=None, objective='binary:logistic', random_state=0,
              reg_alpha=0, reg_lambda=1, scale_pos_weight=1, seed=None,
              silent=None, subsample=1, verbosity=1)

# **Training is done. Time for Prediction.**

In [34]:
y_pred=model.predict(x_test)
acc=(accuracy_score(y_test, y_pred)*100)
print(round(acc,2)) # Printing the Accuracy upto 2 decimal places


94.87


# **We have achieved an accuracy score of 94.87% by using the XGBoost Classifier**